docs/selectors.md

```diff
@@ -72,10 +72,12 @@ XPath engine is equivalent to [`Document.evaluate`](https://developer.mozilla.or
 ### text
 
 Text engine finds an element that contains a text node with passed text. Example: `text=Login`.
-- By default, the match is case-insensitive, and ignores leading/trailing whitespace. This means `text= Login` matches `<button>loGIN </button>`.
-- Text body can be escaped with double quotes for precise matching, insisting on specific whitespace and case. This means `text="Login "` will only match `<button>Login </button>` with exactly one space after "Login".
+- By default, the match is case-insensitive, ignores leading/trailing whitespace and searches for a substring. This means `text= Login` matches `<button>Button loGIN (click me)</button>`.
+- Text body can be escaped with double quotes for precise matching, insisting on exact match, including specified whitespace and case. This means `text="Login "` will only match `<button>Login </button>` with exactly one space after "Login".
 - Text body can also be a JavaScript-like regex wrapped in `/` symbols. This means `text=/^\\s*Login$/i` will match `<button> loGIN</button>` with any number of spaces before "Login" and no spaces after.
 
+> **NOTE** Text engine searches for elements inside open shadow roots, but not inside closed shadow roots or iframes.
```

Comment:
Will we consider it a breaking change if this starts piercing closed shadow dom?

Reply:
Most likely.

Reply:
Minor detail: Isn't this text better left as a generic paragraph/section/note about shadow DOM, and then saying "This is only currently supported by the text engine". Reason is that you can then easily amend with "text engine, css engine" etc.

Reply:
Thanks for pointing this out. I don't think css will ever support shadow dom - that would be against the css spec. We'll probably make attribute selectors pierce shadow dom as well, and I'll make this note more generic.